package.json

```diff
@@ -6,13 +6,18 @@
   "directories": {
     "test": "test"
   },
+  "typings": "chai-immutable.d.ts",
   "scripts": {
     "check-style": "jscs chai-immutable.js test/test.js",
     "test": "npm run test-mocha; npm run test-phantomjs; npm run check-style",
     "test-mocha": "mocha",
     "test-phantomjs": "echo 'mocha-phantomjs is temporarily disabled.' # mocha-phantomjs test/index.html",
     "coverage": "istanbul cover _mocha",
-    "coveralls": "npm run coverage && cat ./coverage/lcov.info | coveralls"
+    "coveralls": "npm run coverage && cat ./coverage/lcov.info | coveralls",
+    "build-typings": "tsc -p .",
+    "test-typings": "npm run build-typings && npm run check-style-typings && mocha typing/build/*.specs.js",
+    "check-style-typings": "tslint typing/lib/**/*.ts typing/test/**/*.ts",
+    "prepublish": "typings install"
```

Comment:
This will run at each npm publish (i.e. every time I push a tag to GitHub that gets picked up by Travis), is that on purpose?